examples/ex1p.cpp

```diff
@@ -52,6 +52,8 @@
 using namespace std;
 using namespace mfem;
 
+#define TEST_PRODUCT
```

Comment:
Revert changes in this file before merging

Reply:
Changes have now been reverted. If a reviewer would like to get my test for ex1p, I can provide it.

Reply:
For the sake of due diligence, please send your test for ex1p.  You can email it to me at my work email address, if that's easiest.  Thanks.

Reply:
Honestly this looks like the sort of thing that could be tested with a unit test.

Reply:
@acfisher I agree that this feature is simple enough that a unit test should cover it, but (being new to mfem testing) I am not sure how to set up a unit test that checks whether a particular MFEM_VERIFY() fails. The feature being added is to prevent users from making the mistake of using iterative mode in a product, which can be difficult to debug. A unit test to ensure this is prevented might set up a product of operators with incorrect modes and check that it fails. I think any test that validates the code (either a unit test or a larger test like the ex1p version I provided) should be sufficient for the PR, but please let me know if you would like to see a unit test added to the PR.

Reply:
I think it's a good opportunity to run you through how the unit tests work.  How about we meet tomorrow to put one together?

Reply:
@acfisher Yes, that sounds good.